src/frameworks/openai/openai-converter.ts

```diff
@@ -0,0 +1,632 @@
+      chunk.type === 'response.completed' &&
+      chunk.response?.output
+    ) {
+      messages.push(
+        ...convertResponsesAPIMessagesToLLMMessages(chunk.response.output)
+      );
+      return null;
+    }
+
+    // Handle new streaming format - text deltas
+    if (chunk.type === 'response.output_text.delta' && chunk.delta) {
+      return {
+        accumulatedContent: chunk.delta,
+        messageRole: 'assistant', // Responses API messages are always from assistant
+      };
+    }
+
+    // Handle new streaming format - completed response (fallback for final message)
+    if (chunk.type === 'response.completed' && chunk.response?.output) {
+      // This handles the case where we have a complete response at the end
+      // but we'll mainly rely on the accumulated deltas above
+      return null;
+    }
+
+    // Handle output item added (establishes the message role)
+    if (chunk.type === 'response.output_item.added' && chunk.item?.role) {
+      return {
+        accumulatedContent: '',
+        messageRole: chunk.item.role,
+      };
+    }
+
+    return null;
+  }
+
+  private async handleChatCompletionsNonStreaming(
+    response: ChatCompletion
+  ): Promise<LLMMessage[]> {
+    response = response as ChatCompletion;
+
+    const messages: LLMMessage[] = [];
+
+    if (!response?.choices) {
+      throw new Error('Invalid chat completions response: missing choices');
+    }
+
+    const firstChoice = response.choices[0];
+    if (firstChoice.message?.role) {
+      const message: LLMMessage = {
+        role: firstChoice.message.role,
+      };
+      if (firstChoice.message?.content) {
+        message.content = firstChoice.message.content;
+      }
+      if (firstChoice.message?.tool_calls) {
+        message.tool_calls = firstChoice.message.tool_calls.map(
+          (tool_call: any) => ({
+            name: tool_call.function.name,
+            arguments: tool_call.function.arguments
+              ? JSON.parse(tool_call.function.arguments)
+              : {},
+            id: tool_call.id,
+          })
+        );
+      }
+      if (message.content || message.tool_calls) {
+        messages.push(message);
+      } else {
+        throw new Error('Invalid response: ' + JSON.stringify(firstChoice));
+      }
+    }
+
+    return messages;
+  }
+
+  private async handleResponseApiNonStreaming(
+    response: Response
+  ): Promise<LLMMessage[]> {
+    const messages: LLMMessage[] = [];
+
+    if (response.output) {
+      messages.push(
+        ...convertResponsesAPIMessagesToLLMMessages(response.output)
+      );
+    } else {
+      throw new Error(
+        'Invalid Responses API response: ' + JSON.stringify(response)
+      );
+    }
+
+    return messages;
+  }
+}
+
+// OpenAI-specific function to convert Responses API messages to LLM messages
+function convertResponsesAPIMessagesToLLMMessages(
+  messages: any[]
+): LLMMessage[] {
+  const extracted_messages: LLMMessage[] = [];
```

Comment:
Please rename to keep typescript conventions.
```suggestion
  const extractedMessages: LLMMessage[] = [];
```